src/persist-client/src/internal/machine.rs

```diff
@@ -111,7 +111,21 @@ where
     }
 
     pub async fn add_rollup_for_current_seqno(&mut self) -> RoutineMaintenance {
-        let rollup = self.applier.write_rollup_blob(&RollupId::new()).await;
+        let state = self.applier.clone_state_for_rollup();
+        let rollup = self
+            .applier
+            .state_versions
+            .write_rollup_for_state(
+                self.applier.shard_metrics.as_ref(),
+                &state,
+                &RollupId::new(),
+            )
+            .await;
+
+        let Some(rollup) = rollup else {
+            return RoutineMaintenance::default();
+        };
+
         let (applied, maintenance) = self.add_rollup((rollup.seqno, &rollup.to_hollow())).await;
```

Comment:
i realized that if we want to efficiently be able to search around old rollups to serve a snapshot at some historical as_of, it'd be super helpful to inline the since and upper next to the seqno and path in the registration. seems worthwhile to add that while you're in here

Reply:
I'd prefer to do this as a follow-up PR. it's not super complicated but it feels logically separated from the rest